### PR TITLE
[SIS-083] Lesson Plan Editor backend

### PR DESCRIPTION
**Pull request by kurtislaw**
# Description

Implemented getLessonPlan and deleteLessonPlan.

Also modified checkFileExists in Local.js slightly. Previous implementation would print FILE EXISTS irregardless of whether the file existed or not. Added a simple if-else to give correct response.

Ticket #

## Type of change

Please delete options that are not relevant.

- [] New feature (non-breaking change which adds functionality)
- [] Bug fix (non-breaking change which fixes an issue)
# Checklist:

- [x] My PR title is formatted as [SIS-\<issue number\>] \<issue name\> (eg. [SIS-010] Create UI button)
- [x] I have linked my issue/ticket on the project board to this PR
- [x] I have sent my PR to #team-sistema-prs on Slack and requested two people to review
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ import LibraryNavIcon from './assets/libraryNavIcon.svg';
 import HomeNavIcon from './assets/homeNavIcon.svg';
 import LessonPlanEditorNavIcon from './assets/lessonPlanEditorNavIcon.svg';
 
+// Imports for the lessonplanservice testing
```
Comment (ramyzhang) [inline App.js]: pls remove tests in App.js!

File: src/services/routes/Local.js [hunk 2]
```diff
@@ -90,7 +90,11 @@ export async function moveFile(oldpath, newpath) {
 export async function checkFileExists(path) {
   return RNFS.exists(path)
     .then(result => {
-      console.log('FILE EXISTS');
+      if (result) {
```
Comment (ramyzhang) [inline src/services/routes/Local.js:108]: great catch!

**Review by ramyzhang — CHANGES_REQUESTED** [review]
pls make these small changes, solve ur merge conflicts, and it'll be ready for emily to work on!! thanks for the great work kurtis!

File: src/services/LessonPlanService.js [hunk 3]
```diff
@@ -12,13 +13,24 @@ const LessonPlanService = {
...
   deleteLessonPlan: async function (name) {
     try {
+      let favouritedPath = `${MAINDIRECTORY}/Favourited/${name}/`;
+      let defaultPath = `${MAINDIRECTORY}/Default/${name}/`;
+      let path;
+
+      // check if file exists, assigning appropriate path if so
+      if (await checkFileExists(favouritedPath)) {
+        path = favouritedPath;
+      } else if (await checkFileExists(defaultPath)) {
```
Comment (Lemiex) [inline src/services/LessonPlanService.js:33]: Right, it is impossible that the lesson plan is in both favouritedPath and defaultPath

**Review by Lemiex — APPROVED** [review]
Everything looks good!

**Pull request merged**